src/devices/bus/centronics/epson_lx810l.cpp

```diff
@@ -409,6 +398,9 @@ uint8_t epson_lx810l_device::porta_r(offs_t offset)
 
 	LOG("%s: lx810l_PA_r(%02x): result %02x\n", machine().describe_context(), offset, result);
 
+	// Update the printhead display, only put this here since this routine gets called frequently
+	m_bitmap_printer->set_printhead_color( m_e05a30->ready_led() ? 0x55ff55 : 0x337733, 0x0);
```

Comment:
This should really be tied to whatever sets the ready LED directly, and also the colors should be defined by the bitmap_printer device itself rather than the callers.  Something like m_bitmap_printer->set_printhead_ready(true);

Reply:
One way to do this would be to make a callback from the bitmap_printer_device::screen_update_bitmap routine that would update the leds.

uint32_t bitmap_printer_device::screen_update_bitmap(screen_device &screen,
							 bitmap_rgb32 &bitmap, const rectangle &cliprect)

Another would be to set the state directly from the e05a30 callbacks in epson_lx810.h, 

	int ready_led() { return !m_centronics_busy; }


	DECLARE_WRITE_LINE_MEMBER(e05a30_centronics_ack) { output_ack(state); }
	DECLARE_WRITE_LINE_MEMBER(e05a30_centronics_busy) { output_busy(state);
                            m_bitmap_printer->set_printhead_ready(!state); }
	DECLARE_WRITE_LINE_MEMBER(e05a30_centronics_perror) { output_perror(state); }
	DECLARE_WRITE_LINE_MEMBER(e05a30_centronics_fault) { output_fault(state); }
	DECLARE_WRITE_LINE_MEMBER(e05a30_centronics_select) { output_select(state); }




I was thinking that the center printhead block would be the green ready LED and the border of the printhead block would be a red "error" LED (like paper error).  

Allowing the caller to set the colors allows a driver like the silentype to use a different color scheme (as it doesn't have any leds).  I liked a two tone beige printhead color for the silentype which kinda matched the silentype's exterior colors (though the actual printhead is white plastic).  Perhaps the caller could set the printhead ready and error colors and on would be the color specified, off be 1/2 or 1/4 of the intensity of the color.